Happy-Algorithms-League/hal-cgp
# Add gallery of examples to documentation

This addresses one point in #174 : It adds a gallery showcasing our examples to the documentation. 

This is achieved in two steps:
- Add the necessary infrastructure (additional `sphinx-gallery` package, some changes in the documentation)
- Restructure example scripts to add comments that comply with the restructured Text formatting of `sphinx-gallery` (see here for a guide: https://sphinx-gallery.github.io/stable/syntax.html). Note that I dissolved the main function in some of the examples because it didn't seem necessary and without putting the main loop in a function, we can nicely break up the code in smaller chunks with docstrings in between.

To test this, execute the following:
```shell
cd docs
make html
firefox _build/html/index.html &
```
and navigate to `examples`. 


*Out of scope of this PR:*
- inclusion of figures from the examples: see comment in the issue #174 .
